src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs

```diff
@@ -0,0 +1,4224 @@
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_Overload_NamedTokenUnordered()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync(s: ""Hello world"", c: CancellationToken.None, x: 5);
+    }
+    Task MethodAsync(int x, string s) => Task.CompletedTask;
+    Task MethodAsync(int x, string s, CancellationToken c) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_CancellationTokenSource_ParamsUsed_Order1()
+        {
+            /*
+            CancellationTokenSource has 3 different overloads that take CancellationToken arguments.
+            We should detect if a ct is passed and not offer a diagnostic, because it's considered one of the `params`.
+
+            public class CancellationTokenSource : IDisposable
+            {
+                public static CancellationTokenSource CreateLinkedTokenSource(CancellationToken token);
+                public static CancellationTokenSource CreateLinkedTokenSource(CancellationToken token1, CancellationToken token2);
+                public static CancellationTokenSource CreateLinkedTokenSource(params CancellationToken[] tokens);
+            }
+            */
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(CancellationToken token){}
+    public static void Method(CancellationToken token1, CancellationToken token2){}
+    public static void Method(params CancellationToken[] tokens){}
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_CancellationTokenSource_ParamsUsed_Order2()
+        {
+            /*
+            CancellationTokenSource has 3 different overloads that take CancellationToken arguments.
+            We should detect if a ct is passed and not offer a diagnostic, because it's considered one of the `params`.
+
+            public class CancellationTokenSource : IDisposable
+            {
+                public static CancellationTokenSource CreateLinkedTokenSource(CancellationToken token);
+                public static CancellationTokenSource CreateLinkedTokenSource(CancellationToken token1, CancellationToken token2);
+                public static CancellationTokenSource CreateLinkedTokenSource(params CancellationToken[] tokens);
+            }
+            */
+            return CS8VerifyAnalyzerAsync(@"
+using System.Threading;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        CTS.Method(ct); // Don't diagnose
+    }
+}
+class CTS
+{
+    public static void Method(CancellationToken token){}
+    public static void Method(params CancellationToken[] tokens){}
+    public static void Method(CancellationToken token1, CancellationToken token2){}
```

Comment:
I don't understand how this test differs from previous one? The order in which overloads are defined in source should not affect overload resolution.

Reply:
I think I was going to test different orderings and I ended up deciding against it, but I forgot to remove the extra methods. I'll remove them.

Reply:
Oh actually yes, the overloads are in different orders. I'll remove them anyway.